test/test_quantization.py

```diff
@@ -831,31 +831,55 @@ def test_tensor_observer_scriptable(self, qdtype, qscheme):
         x = torch.tensor([1.0, 2.0, 2.0, 3.0, 4.0, 5.0, 6.0])
         y = torch.tensor([4.0, 5.0, 5.0, 6.0, 7.0, 8.0])
         myobs(x)
         myobs(y)
-        self.assertEqual(myobs.min_val, -1.5)
-        self.assertEqual(myobs.max_val, 8.5)
-        self.assertEqual(myobs.histogram, [0., 0., 1., 2., 1., 2., 3., 2., 1., 1.])
+        if use_fixed_width is True:
+            self.assertEqual(myobs.min_val, -1.5)
+            self.assertEqual(myobs.max_val, 8.5)
+            self.assertEqual(myobs.histogram, [0., 0., 1., 2., 1., 2., 3., 2., 1., 1.])
+        else:
+            self.assertEqual(myobs.min_val, 1.0)
+            self.assertEqual(myobs.max_val, 8.0)
+            self.assertEqual(myobs.histogram, [1., 2., 0., 1., 2., 2., 2., 1., 1., 1.])
+
         qparams = myobs.calculate_qparams()
-        if reduce_range:
-            if qscheme == torch.per_tensor_symmetric:
-                ref_scale = 0.066666 * 255 / 127
-                ref_zero_point = 0 if qdtype is torch.qint8 else 128
+
+        if use_fixed_width is True:
+            if reduce_range:
+                if qscheme == torch.per_tensor_symmetric:
+                    ref_scale = 0.066666 * 255 / 127
+                    ref_zero_point = 0 if qdtype is torch.qint8 else 128
+                else:
+                    ref_scale = 0.0333333 * 255 / 127
+                    ref_zero_point = -64 if qdtype is torch.qint8 else 0
             else:
-                ref_scale = 0.0333333 * 255 / 127
-                ref_zero_point = -64 if qdtype is torch.qint8 else 0
+                if qscheme == torch.per_tensor_symmetric:
+                    ref_scale = 0.066666
+                    ref_zero_point = 0 if qdtype is torch.qint8 else 128
+                else:
+                    ref_scale = 0.0333333
+                    ref_zero_point = -128 if qdtype is torch.qint8 else 0
         else:
-            if qscheme == torch.per_tensor_symmetric:
-                ref_scale = 0.066666
-                ref_zero_point = 0 if qdtype is torch.qint8 else 128
+            if reduce_range:
+                if qscheme == torch.per_tensor_symmetric:
+                    ref_scale = 0.062745 * 255 / 127
+                    ref_zero_point = 0 if qdtype is torch.qint8 else 128
+                else:
+                    ref_scale = 0.0313725 * 255 / 127
+                    ref_zero_point = -64 if qdtype is torch.qint8 else 0
             else:
-                ref_scale = 0.0333333
-                ref_zero_point = -128 if qdtype is torch.qint8 else 0
+                if qscheme == torch.per_tensor_symmetric:
+                    ref_scale = 0.062745
+                    ref_zero_point = 0 if qdtype is torch.qint8 else 128
+                else:
+                    ref_scale = 0.0313725
+                    ref_zero_point = -128 if qdtype is torch.qint8 else 0
+
```

Comment:
Do you have a test to ensure that L2 observer is scriptable?

Reply:
Can I make it scriptable in another PR? Since making it scriptable is not trivial and could add extra code that complicate the logic of this PR.

Reply:
I don't think the observer itself necessarily needs to be scriptable. We can totally call python objects back from script and I think the quantization process would happen in python environment anyway